zone.go

```diff
@@ -0,0 +1,244 @@
+package dns
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+)
+
+// AddDomain adds origin to s if s is not already a FQDN.
+func AddDomain(origin, s string) string {
```

Comment:
Logically one can't know if utility functions like IsEdns0, AddDomain, TrimDomain are used by project outside of this module.  I agree that documenting them would help people find them.  I know that getting started using the module was difficult for me because the documentation was so brief.

Also, while a function such as IsEdns0() may seem obvious to some, the minutiae of DNS is staggeringly huge. Such utility functions prevent stupid bugs from haunting the internet.  For example without CountLabel, most people would just use strings.Count(s, "."), without IsSubDomain most people would just use strings.IsSubDomain(s, domain).

That said...

It sounds to me like this project only wants contributions in a very specific domain of functionality.  I'm not sure that that domain is, but I think it would make the project easier to contribute to if that domain was described in the README.md.  Feel free to contact me off-line if you'd like to know what I think it is, based on reverse engineering the comments posted on this PR.